pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -115,16 +117,18 @@ protected void scheduleReadOnActiveConsumer() {
         }
 
         readOnActiveConsumerTask = topic.getBrokerService().executor().schedule(() -> {
-            if (log.isDebugEnabled()) {
-                log.debug("[{}] Rewind cursor and read more entries after {} ms delay", name,
-                        serviceConfig.getActiveConsumerFailoverDelayTimeMillis());
+            // in order to prevent redeliverUnacknowledgedMessages rewind cursor again
+            synchronized (PersistentDispatcherSingleActiveConsumer.this) {
```

Comment:
Could you please provide more context here? I'm more sure why adding `synchronized` here can prevent rewinding the cursor again. The Runnable instance waiting here but after the previous one is complete, it will rewind the cursor.